### PR TITLE
More `from_str` cleanups

### PR DESCRIPTION
**Pull request by robertbastian**
#4931

### CODE REVIEW

**Review by sffc — DISMISSED** [review]
You're removing impls from TinyStr wrapper types and instead using direct construction, which seems like a positive change since it avoids an unwrap, despite being more verbose.

File: components/plurals/src/operands.rs [hunk 1]
```diff
@@ -112,34 +110,26 @@ pub enum OperandsError {
...
-    if let Some((base, exponent)) = input.split_once('e') {
-        Ok((base, exponent.parse()?))
-    } else {
-        Ok((input, 0))
-    }
-}
-
-impl FromStr for PluralOperands {
+#[cfg(feature = "datagen")]
+impl core::str::FromStr for PluralOperands {
```
[the remark this answers is not in the document]
Comment (robertbastian) [reply components/plurals/src/operands.rs:121]: Users can create plural operands from all numeric types as well as fixed decimal, infallibly.

**Review by Manishearth — APPROVED** [review]

File: ffi/ecma402/src/pluralrules.rs [hunk 2]
```diff
@@ -109,11 +109,8 @@ pub(crate) mod internal {
     /// Converts the number to format into the operands representation.
     pub fn to_icu4x_operands(n: f64, opts: Options) -> PluralOperands {
         dbg!("n={}", n);
-        let nstr = fixed_format(n, &opts);
-        #[allow(clippy::unwrap_used)] // TODO(#1668) Clippy exceptions need docs or fixing.
-        let ret = nstr.parse().unwrap();
-        dbg!("ret={:?}\n---\n", &ret);
-        ret
+        let _nstr = fixed_format(n, &opts);
+        todo!()
```
[the remark this answers is not in the document]
Comment (robertbastian) [reply ffi/ecma402/src/pluralrules.rs:113]: what do you want me to do here? I'm tempted to just remove the whole crate

[the remark this answers is not in the document]
Comment (sffc) [reply ffi/ecma402/src/pluralrules.rs:113]: I think endeavor to keep it working. It's not that much code to keep working.

In this case, just change `PluralOperands::from_str` to `FixedDecimal::from_str`.

**Review by sffc — CHANGES_REQUESTED** [review]

File: ffi/ecma402/src/pluralrules.rs [hunk 3]
```diff
@@ -127,105 +39,6 @@ pub(crate) mod internal {
...
-                    t: 5,
-                    c: 0,
-                }
-                .into(),
-            }];
-            for test in tests {
-                let actual = to_icu4x_operands(test.n, test.opts.clone());
-                assert_eq!(test.expected, actual, "test: {:?}", &test);
-            }
-        }
```
Comment (sffc) [inline ffi/ecma402/src/pluralrules.rs:227]: Nit: Please restore the test for `to_icu4x_operands`

**Review by sffc — APPROVED** [review]

**Review by sffc — APPROVED** [review]

**Pull request merged**